integration-tests/resteasy-reactive-rest-client-multipart/src/main/java/io/quarkus/it/rest/client/multipart/MultipartClient.java

```diff
@@ -0,0 +1,119 @@
+package io.quarkus.it.rest.client.multipart;
+
+import java.io.File;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+import org.jboss.resteasy.reactive.MultipartForm;
+import org.jboss.resteasy.reactive.PartType;
+
+import io.vertx.core.buffer.Buffer;
+
+@Path("/echo")
+@RegisterRestClient(configKey = "multipart-client")
+public interface MultipartClient {
+
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.TEXT_PLAIN)
+    @Path("/binary")
+    String sendByteArrayAsBinaryFile(@MultipartForm WithByteArrayAsBinaryFile data);
+
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.TEXT_PLAIN)
+    @Path("/binary")
+    String sendBufferAsBinaryFile(@MultipartForm WithBufferAsBinaryFile data);
+
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.TEXT_PLAIN)
+    @Path("/binary")
+    String sendFileAsBinaryFile(@MultipartForm WithFileAsBinaryFile data);
+
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.TEXT_PLAIN)
+    @Path("/text")
+    String sendByteArrayAsTextFile(@MultipartForm WithByteArrayAsTextFile data);
+
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.TEXT_PLAIN)
+    @Path("/text")
+    String sendBufferAsTextFile(@MultipartForm WithBufferAsTextFile data);
+
+    @POST
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.TEXT_PLAIN)
+    @Path("/text")
+    String sendFileAsTextFile(@MultipartForm WithFileAsTextFile data);
+
+    class WithByteArrayAsBinaryFile {
+
+        @FormParam("file")
+        @PartType(MediaType.APPLICATION_OCTET_STREAM)
+        public byte[] file;
```

Comment:
We'll have to be very careful on how we document this and other blocking variants

Reply:
I updated the PR and, among others, added some basic documentation. Could you elaborate on ^? What do you think is worth mentioning?